straxen/matplotlib_utils.py

```diff
@@ -128,13 +141,19 @@ def plot_on_single_pmt_array(
     else:
         ax.set_axis_off()
     if dead_pmts is not None:
-        _dead_mask = [pi in dead_pmts for pi in pos['i']]
-        result = plt.scatter(
-            pos[_dead_mask]['x'],
-            pos[_dead_mask]['y'],
-            c=dead_pmt_color,
-            **kwargs)
-
+        for p in pos:
+            if p['i'] in dead_pmts:
+                plt.text(p['x'], p['y'], str(p['i']),
+                         horizontalalignment='center',
+                         verticalalignment='center',
+                         fontsize=dead_pmt_label_size,
+                         color=dead_pmt_label_color)
+                ax.add_artist(plt.Circle((p['x'], p['y']), 3.75,
```

Comment:
I have defined the radius 3.75 because visually it fits but if you know ho to make it general give the figure size it will be perfect